spec/lib/catalog/rbac/access_spec.rb

```diff
@@ -175,16 +159,23 @@
     it_behaves_like "resource checking", :resource_check, ["order", "321", Portfolio], "order", Portfolio, :has_order_permission
   end
 
-  describe "#update_access_check" do
-    it_behaves_like "resource checking", :update_access_check, [], "update", PortfolioItem, :has_update_permission
-  end
+  describe "helper methods" do
+    let(:subject) { described_class.new(user_context, portfolio_item) }
+    let(:rbac_enabled) { true }
+    it "#update_access_check" do
+      expect(subject).to receive(:resource_check).with('update')
+      subject.update_access_check
+    end
 
-  describe "#read_access_check" do
-    it_behaves_like "resource checking", :read_access_check, [], "read", PortfolioItem, :has_read_permission
-  end
+    it "#read_access_check" do
+      expect(subject).to receive(:resource_check).with('read')
+      subject.read_access_check
+    end
 
-  describe "#destroy_access_check" do
-    it_behaves_like "resource checking", :destroy_access_check, [], "delete", PortfolioItem, :has_delete_permission
+    it "#destroy_access_check" do
+      expect(subject).to receive(:resource_check).with('delete')
+      subject.destroy_access_check
+    end
```

Comment:
@eclarizio Without those changes this spec cannot be tested because of the assumptions that it makes notably PortfolioItems exist in the AccessControlEntry table. resource_check and permission_check are the only main function here the rest are all helper functions.

Reply:
We should probably change `subject` in those describes to then just be `subject { described_class.new(user_context, portfolio) }` and create a `portfolio` to ensure that it works properly.

My suggestion: https://gist.github.com/eclarizio/8308467601f0e2bb373e3d0c7cf11cf6